### DISCUSSION TITLE
Novice question to understand how strategy.matrix works

### DISCUSSION THREAD

**Discussion by aaronware** [community/community#25781] [Actions]
<p>Can a strategy.matrix be a list of anything that a job can loop through?. For example, I have a WordPress project that has at least one, but possibly multiple themes within it. If a theme exists, I need to have that theme run a “build” process, using a few yarn commands. Currently, I am targeting each individual folder for the project but it doesn’t really scale and isn’t reusable.</p>
<p>I was trying to make this more of a reusable workflow but in order to do that I want to allow for an input of something like</p>
<ul>
<li>theme: [ theme-1, theme-2, theme-3 ]</li>
</ul>
<p>and based on that matrix have the job run the same process for every theme in the matrix. But in running this process it told me my workflow was invalid.</p>
<p>Does anyone have a recommendation on how to accomplish this or is it not the proper approach.</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>Looping over a list would be the easiest possibly use case for a matrix, that should work without issue, like this:</p>
<pre><code class="lang-yaml">strategy:
  matrix:
    theme:
      - theme-1
      - theme-2
      - theme-3
</code></pre>
<p>How are you trying to combine that with a reusable workflow? Do you want to create the matrix in a job of the resuable workflow from an input to that workflow?</p>

**Comment by aaronware** [comment]
<p><a class="mention" href="/airtower-luna">@airtower-luna</a> thanks for the quick response! I actually think the issue wasn’t the matrix, but how I was trying to pass a list of themes to the reusable workflow as an input from my project workflow</p>
<p>In the project workflow, I had simply treated the input with an array similar to how most matrix examples work.</p>
<p><code>themes: [ theme-1, theme-2, theme-3 ]</code></p>
<p>I think the problem is that what I have is an invalid yaml format within my project workflow. To your point, I should probably make my input similar to what you have (which is very readable), or as a stringified JSON, I suppose. And then either just get the array, like you showed, or use fromJSON() as an expression within the reusable workflow.</p>
<p>Thanks for giving a shove in the right direction</p>

**Comment by airtower-luna** [comment]
<p>Stringified JSON is probably the way to go, AFAIK inputs are strings and that’s it, no data structures supported. I’ve seen people people discuss here how they used <code>fromJSON()</code> to define a matrix so I’m pretty sure that works, even though I haven’t tried it myself.</p>

**Comment by aaronware** [comment]
<p>I went with the JSON approach as it allowed more flexibility later on if needed. If anyone is wondering. It appears the matrix will loop through any values in order to do any subsequent steps in a workflow which is really useful.</p>
